docs/calendar.md

```diff
@@ -347,6 +347,39 @@ Temporal.Calendar.from('iso8601').fields(['month', 'day']);
 ```
 <!-- prettier-ignore-end -->
 
+### calendar.**mergeFields**(_fields_: object, _additionalFields_: object) : object
+
+**Parameters:**
+
+- `fields` (object): A plain object with properties representing calendar units.
+- `additionalFields` (object): Another plain object with properties representing calendar units.
+
+**Returns:** the same object that was passed in as `fields`, with its properties modified.
```

Comment:
why mutate, rather than returning a new object? The naive implementation would be `return { ...fields, ...additionalFields };`

Reply:
Good question; before Temporal calls this method, we have already copied all the properties onto a new object with ToTemporalDateFields etc. Returning a new object here would create a second new object unnecessarily.

Reply:
Hmm. Mutation is widely considered a bad practice, and potentially deoptimizes object access as well. I don't think the builtins should be worried about creating a new object "unnecessarily", especially since in the common case, engines will be able to silently optimize away the new object.

Reply:
Fair enough, Justin brought up the same question but I hadn't taken into account that you can optimize out the call completely if it's a builtin calendar and Temporal.Calendar.prototype is unchanged. I'll change this.